lib/tasks/elasticsearch.rake

```diff
@@ -1,15 +1,41 @@
 namespace :elasticsearch do
-  desc 'Deletes the "resource" index and regenerates it with all records currently in Resource'
-  task reset_resource_index: :environment do
+  def reset_index(klass)
     client = Elasticsearch::Client.new(
       url: ENV.fetch('BONSAI_URL', 'http://localhost:9200'), log: true
     )
 
-    if client.indices.exists? index: Resource.index_name
-      client.indices.delete index: Resource.index_name
+    if client.indices.exists? index: klass.index_name
+      client.indices.delete index: klass.index_name
     end
 
-    Resource.__elasticsearch__.create_index!
-    Resource.import
+    klass.__elasticsearch__.create_index!
+
+    begin
+      klass.import
+    rescue Faraday::ConnectionFailed => e
```

Comment:
Interesting. Is Faraday used to connect to ES?

Does indexing one at a time resolve this? More curious than anything else.